src/SprykerSdk/Zed/ComposerConstrainer/Business/Finder/UsedForeignModuleFinder/UsedForeignModuleFinder.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Zed\ComposerConstrainer\Business\Finder\UsedForeignModuleFinder;
+
+use Closure;
+use Generated\Shared\Transfer\UsedModuleCollectionTransfer;
+use Generated\Shared\Transfer\UsedModuleTransfer;
+use ReflectionClass;
+use SprykerSdk\Zed\ComposerConstrainer\Business\Composer\ComposerJson\ComposerJsonReaderInterface;
+use SprykerSdk\Zed\ComposerConstrainer\Business\Finder\FinderInterface;
+use SprykerSdk\Zed\ComposerConstrainer\ComposerConstrainerConfig;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+class UsedForeignModuleFinder implements FinderInterface
+{
+    /**
+     * @var \SprykerSdk\Zed\ComposerConstrainer\Business\Composer\ComposerJson\ComposerJsonReaderInterface
+     */
+    protected $composerJsonReader;
+
+    /**
+     * @var \SprykerSdk\Zed\ComposerConstrainer\ComposerConstrainerConfig
+     */
+    protected $config;
+
+    /**
+     * @param \SprykerSdk\Zed\ComposerConstrainer\ComposerConstrainerConfig $config
+     * @param \SprykerSdk\Zed\ComposerConstrainer\Business\Composer\ComposerJson\ComposerJsonReaderInterface $composerJsonReader
+     */
+    public function __construct(ComposerConstrainerConfig $config, ComposerJsonReaderInterface $composerJsonReader)
+    {
+        $this->config = $config;
+        $this->composerJsonReader = $composerJsonReader;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\UsedModuleCollectionTransfer
+     */
+    public function find(): UsedModuleCollectionTransfer
+    {
+        $usedModuleCollectionTransfer = new UsedModuleCollectionTransfer();
+
+        if (!is_dir($this->config->getSourceDirectory())) {
+            return $usedModuleCollectionTransfer;
+        }
+
+        foreach ($this->createFinder() as $splFileInfo) {
+            $usedModuleCollectionTransfer = $this->addUsedForeignModules($usedModuleCollectionTransfer, $splFileInfo);
+        }
+
+        return $usedModuleCollectionTransfer;
+    }
+
+    /**
+     * @return \Symfony\Component\Finder\Finder|\Symfony\Component\Finder\SplFileInfo[]
+     */
+    protected function createFinder(): Finder
+    {
+        $finder = new Finder();
+        $finder
+            ->files()
+            ->in($this->config->getSourceDirectory())
+            ->filter($this->getFileFilter())
+            ->exclude($this->config->getExcludedNamespaces())
+            ->name('*.php');
+
+        return $finder;
+    }
+
+    /**
+     * @return \Closure
+     */
+    protected function getFileFilter(): Closure
```

Comment:
IMO would be nice to name this method more descriptive. It's not so clear what this method should filter

Reply:
It was left as it is intentionally, in sake of consistency (with ExtendedModuleFinder) and reduce amount of modified code.